.github/workflows/build.yaml

```diff
@@ -0,0 +1,478 @@
+name: Build
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    types:
+      - opened
+      - reopened
+      - synchronize
+
+jobs:
+  pre-build-updater:
+    runs-on: ubuntu-latest
+    container:
+      image: quay.io/stackrox-io/apollo-ci:scanner-test-0.3.61
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+          ref: ${{ github.event.pull_request.head.sha }}
+
+      - uses: ./.github/actions/job-preamble
+
+      - name: Cache Go dependencies
+        uses: ./.github/actions/cache-go-dependencies
+
+      - name: Build updater
+        run: make build-updater
+
+      - name: Bundle the build to preserve permissions
+        run: tar -cvzf updater-build.tgz bin/updater
+
+      - uses: actions/upload-artifact@v3
+        with:
+          name: updater-build
+          path: updater-build.tgz
+
+  pre-build-scanner:
+    runs-on: ubuntu-latest
+    container:
+      image: quay.io/stackrox-io/apollo-ci:scanner-test-0.3.61
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+          ref: ${{ github.event.pull_request.head.sha }}
+
+      - uses: ./.github/actions/job-preamble
+
+      - name: Cache Go dependencies
+        uses: ./.github/actions/cache-go-dependencies
+
+      - name: Build Scanner
+        run: make scanner-build-nodeps
+
+      - name: Bundle the build to preserve permissions
+        run: tar -cvzf scanner-build.tgz image/scanner/bin/scanner
+
+      - uses: actions/upload-artifact@v3
+        with:
+          name: scanner-build
+          path: scanner-build.tgz
+
+  style-check:
+    env:
+      ARTIFACT_DIR: /artifacts
+    runs-on: ubuntu-latest
+    container:
+      image: quay.io/stackrox-io/apollo-ci:scanner-test-0.3.61
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+          ref: ${{ github.event.pull_request.head.sha }}
+
+      - uses: ./.github/actions/job-preamble
+
+      - name: Cache Go dependencies
```

Comment:
In a sense, both. If the jobs run with new or changed dependencies, it would create a cache, but it would typically fetch from the cache. `.github/actions/cache-go-dependencies/action.yaml` should have the exact details on how that process works but the reason we need the Go cache in this job is because of the `golangci-lint` make target (actual name is `$(GOLANGCILINT_BIN)`)